### PR TITLE
docs: Use KEYVAULT_URL in quick start guide

### PR DESCRIPTION
**Pull request by karataliu**
**Reason for Change**:
Update doc to use KEYVAULT_URL, so as to support sovereign clouds

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [x] no

**Notes for Reviewers**:
Follow-up #561

### CODE REVIEW

File: docs/book/src/quick-start.md [hunk 1]
```diff
@@ -237,8 +238,8 @@ spec:
     - image: ghcr.io/azure/azure-workload-identity/msal-go
       name: oidc
       env:
-      - name: KEYVAULT_NAME
-        value: ${KEYVAULT_NAME}
+      - name: KEYVAULT_URL
+        value: ${KEYVAULT_URL}
```
Comment (aramase) [inline docs/book/src/quick-start.md:242]: Let's add `KEYVAULT_URL` as an additional environment variable and not remove `KEYVAULT_NAME`. If old version of `msal-go` is used, it'll not work with `KEYVAULT_URL`. We can remove `KEYVAULT_NAME` after a few releases and after every [example](https://github.com/Azure/azure-workload-identity/tree/main/examples) is updated to use `KEYVAULT_URL`.

**Review by aramase — CHANGES_REQUESTED** [review]

**Comment by aramase** [comment]
@karataliu could you also update the commit to say `docs: Use KEYVAULT_URL in quick start guide` as this a documentation update?

File: docs/book/src/quick-start.md (hunk 1 above)
Comment (karataliu) [reply docs/book/src/quick-start.md:244]: updated.

**Review by aramase — APPROVED** [review]

**Pull request merged**